tests/integration-tests/tests/resource_leaks/test_resource_leaks.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from utils import get_compute_nodes_instance_ips
+
+from tests.common.assertions import assert_head_node_is_running
+
+
+@pytest.mark.usefixtures("instance", "os", "scheduler")
+def test_resource_leaks(
```

Comment:
can we avoid having a specific integration test for it? and instead add this logic to check file descriptors and resource leak by default in all the tests. I mean, we don't need to wait 30 minutes in every test, but we can do the check.

Reply:
I update the PR to add the logic in functions in utils.py and called it in the starccm test. Unfortunately I don't think we can add it by default to all the tests unless we add the assert in cluster creation/deletion, which I think we'd want to avoid

Reply:
How long does the starccm test run?

Reply:
~20 mins, which is long enough for the resource leak test to fail. I tested running this starccm test with the resource leak issue [here](https://github.com/aws/aws-parallelcluster-node/pull/421) and it failed as expected.